patches/minecraft/net/minecraft/item/ItemArmor.java.patch

```diff
@@ -41,8 +53,30 @@
 +                case GOLD:    return Items.field_151043_k;
 +                case IRON:    return Items.field_151042_j;
 +                case DIAMOND: return Items.field_151045_i;
-+                default:      return customCraftingMaterial;
++                default:      return Items.field_190931_a;
 +            }
          }
  
          @SideOnly(Side.CLIENT)
+@@ -281,5 +305,21 @@
+         {
+             return this.field_189417_k;
+         }
++
++        public ArmorMaterial setRepairItem(ItemStack stack)
++        {
++            if (!this.repairMaterial.func_190926_b()) throw new RuntimeException("Repair material has already been set");
++            if (this == LEATHER || this == CHAIN || this == GOLD || this == IRON || this == DIAMOND) throw new RuntimeException("Can not change vanilla armor repair materials");
++            this.repairMaterial = stack;
++            return this;
++        }
++
++        public ItemStack getRepairItemStack()
++        {
++            if (!repairMaterial.func_190926_b()) return repairMaterial;
++            Item ret = this.func_151685_b();
++            repairMaterial = new ItemStack(ret,1,net.minecraftforge.oredict.OreDictionary.WILDCARD_VALUE);
++            return repairMaterial;
```

Comment:
We should return `repairMaterial.copy()` to make sure that the repair material doesn't get made invalid accidentally.